docs/editor/testing.md

```diff
@@ -0,0 +1,173 @@
+
+After running a test, the editor gutter displays the test status.
+
+When you run tests, the test result status is displayed in the Test Explorer view and editor gutter. The test results are color-coded to indicate the status of the test (pass or fail). When a test fails, notice that the test error message is shown as an overlay in the editor.
+
+Use the **Show Output** button in the Test Explorer to view the test output in the **Test Results** panel.
+
+![Test Results panel](images/testing/test-results-panel.png)
+
+You can access the commands for the Test Explorer, such as running or debugging tests, via the Command Palette (`kb(workbench.action.showCommands)`). Enter `Test Explorer:` to find the corresponding commands.
+
+## Test coverage
+
+Test coverage is a measure of how much of your code is covered by your tests. It helps you identify areas of your code that are not being tested. VS Code provides built-in support for running tests with coverage and viewing the coverage results.
+
+Similar to running and debugging tests, you can run tests with coverage by using the actions in the Test Explorer view, editor gutter, or commands in the Command Palette (`kb(workbench.action.showCommands)`).
+
+![Run tests with coverage](images/testing/run-tests-with-coverage.png)
+
+After you run tests with coverage, you can view the coverage results in different locations:
+
+- In the Test Coverage view
+
+    A tree view shows the tests with their coverage percentage. A color indicator also gives a visual cue about the coverage percentage. Hover over each line to see more details about the coverage results.
+
+    ![Test Coverage view](images/testing/test-coverage-view.png)
+
+- As an overlay in the editor
+
+    For code files that have test coverage, the editor shows a color overlay in the gutter to indicate which lines are covered by tests or not. When you hover over the gutter, notice that for covered lines, there is also an indicator for the number of times the line was executed.
+
+    ![View coverage in editor](images/testing/view-coverage-in-editor.png)
+
+- In the Explorer view, which shows the coverage percentage of your code files
+
+    The Explorer view shows the coverage percentage of your code files. Hover over a file or node in the Explorer to see more details about the coverage results.
+
+    ![View coverage in Explorer view](images/testing/view-coverage-in-explorer.png)
+
+- In the diff editor
+
+    If you have a diff editor open, the coverage results are also shown in the diff editor, similar to how they're shown in the editor.
+
+## Task integration
+
+Tasks in VS Code can be configured to run scripts and start processes within VS Code, without having to enter a command line or write new code. In VS Code, you can define a default test task that runs your tests, and optionally create keyboard shortcuts to run the tests.
```

Comment:
Currently tasks don't have special integration into VS Code's testing functionality, so running tests in a task won't update VS Code's UI.

Reply:
Included this in the note that extension can implement integrated test automation features.